src/session_state.rs

```diff
@@ -2,26 +2,45 @@
 ///
 ///
 use embedded_nal::IpAddr;
+use embedded_time::{duration::Seconds, Clock, Instant};
 use heapless::{consts, String, Vec};
 
-pub struct SessionState {
+pub struct SessionState<C>
+where
+    C: Clock,
+{
     pub connected: bool,
-    pub keep_alive_interval: u16,
+
+    /// (embedded_time doesn't support u16.)
+    pub keep_alive_interval: Seconds<u32>,
```

Comment:
There's really no need to bring the clock down into the session state. The `SessionState` is an MQTT abstraction and uses well-defined units for values. Let's keep the `embedded_time::Clock` out of this layer.